### ISSUE TITLE
Consider to use NodaTime

### ISSUE THREAD

**Issue by hikalkan** [abpframework/abp#263]
It has especially good utilities for time, time zone... etc.

**Comment by hikalkan** [comment]
I am moving this to v3.0 to **consider** to include to the framework, as we can make a breaking change with v3.0 if necessary.

**Comment by evrame** [comment]
Hi there.

I am trying to use NodaTime with Abp with no luck being able to serialize.
I have a test app service like so:

`
[AllowAnonymous]
        public async Task PostTestNoda(LocalDate date)
        {
         ... some code
        }
`

And here is my configuration in the module:

`
Configure<JsonSerializer>(options =>
            {
                options.ConfigureForNodaTime(DateTimeZoneProviders.Tzdb);
                options.Converters.Add(NodaConverters.LocalDateConverter);
            });
`

However I am ending up with a default LocalDate object being passed to TestNoda.

Any idea where I might have missed something?
Is there a timeline on when you would implement NodaTime in Abp?

Thanks

**Comment by jhulbertpmn** [comment]
Very interested in this as well.
I did a little test replacing the default clock service in abp with nodatime implementations and had some success but there were parts which felt a little hacky. It seems like there may need to be some changes to the timing interfaces in abp to implement this fully

**Comment by GMCfourX4** [comment]
I would like to see this done as well.

**Comment by tvddev** [comment]
same for us.

**Comment by spospisil** [comment]
Having an implementation for this is critical for our business as we have a scheduling component to our enterprise software product that if regionality/time zone differences are not accurately factored in, appointments date/times will not be correct.

Any update on when this can be done?

**Comment by Alexbits** [comment]
Yes, Nodatime is much superior to DateTime. I have a project with time zones, and this is critical.

**Comment by GMCfourX4** [comment]
Any updates? This would be a very helpful add-on for anyone who deals with records which relate to the future.

**Comment by GMCfourX4** [comment]
Still hopeful that we see a "formal" ABP implementation
